dask/dataframe/core.py

```diff
@@ -3038,6 +3074,33 @@ def drop_columns(df, columns, dtype):
     return df
 
 
+def idxmaxmin_chunk(x, **kwargs):
+    fn = kwargs['fn']
+    axis = kwargs['axis']
+    idx = getattr(x, fn)(axis=axis)
+    minmax = 'max' if fn == 'idxmax' else 'min'
+    value = getattr(x, minmax)(axis=axis)
+    n = len(x)
+    if isinstance(idx, pd.Series):
+        return pd.DataFrame({'idx': idx, 'value': value, 'n': [n] * len(idx)}).T
+    return pd.DataFrame({'idx': [idx], 'value': [value], 'n': [n]}).T
+
+def idxmaxmin_row(x, fn=None):
+    idx = x.ix['idx'].reset_index(drop=True) # .astype(int)
+    try:
+        idx = idx.astype(int)
+    except:
+        pass
```

Comment:
Are you referring to whats stored in the `idx` column?  If so these are the index values. This only works in the case of known_locations.  This would have to be modified in the case of unknown locations and do the index math. 

As for the try/except, yea its dangerous, was an interim solution, as I couldn't think of a way to set the dtype of the idx column in `idxmaxmin_chunk`.